File: Problem_Sets/HackerRank/TimeConversion/TimeConversion.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'timeConversion' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING s as parameter.
#

def timeConversion(s):
    # Write your code here
    ampm = s[8:]
    if ampm == 'AM':
        if s[:2] == '12':
            return '00'+s[2:8]
        else:
            return s[:8]
    elif ampm == 'PM':
        if int(s[:2]) < 12:
            return str(int(s[:2])+12) + s[2:8]
        else:
            return s[:8]

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = timeConversion(s)

    fptr.write(result + '\n')

    fptr.close()

File: Problem_Sets/HackerRank/TimeConversion/TimeConversion.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * Complete the 'timeConversion' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

string timeConversion(string s) {
    int hrs = stoi(s.substr(0, 2));
    char amPm = s[8];
    if (amPm == 'P' && hrs != 12) {
        hrs += 12;
    }
    else if (amPm == 'A' && hrs == 12) {
        hrs = 0;
    }
    string convertedHr = (hrs < 10) ? "0" + to_string(hrs) : to_string(hrs);
    
    stringstream ss;
    ss << convertedHr << ":" << s.substr(3,2) << ":" << s.substr(6,2);
    return ss.str();
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = timeConversion(s);

    fout << result << "\n";

    fout.close();

    return 0;
}
